### ISSUE TITLE
[Bug]: focus indicators don't work in Shadow DOM [v9]

### ISSUE THREAD

**Issue by layershifter** [microsoft/fluentui#27760]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/old-field-hbt64i

### Bug Description

- Open a sandbox (https://codesandbox.io/s/old-field-hbt64i?file=/src/App.js)
- Press <kbd>Tab</kbd>, notice that a button gets focus
  <img width="344" alt="image" src="https://user-images.githubusercontent.com/14183168/236211304-6fd5a875-f6cc-4b70-82ce-a780a6755d8b.png">

## Actual Behavior
A button does not have a focus ring

## Expected Behavior
If <kbd>Tab</kbd> will be pressed once more, a focus ring appears on a button outside Shadow DOM properly
fill this out

<img width="453" alt="image" src="https://user-images.githubusercontent.com/14183168/236211629-75c18259-670f-48bf-b359-b15b3575ab87.png">

### Logs

```shell
N/A
```

### Requested priority

High

### Products/sites affected

N/A

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by layershifter** [comment]
Details on that:
- We use a polyfill for focus visible, that polyfill should be initialized
- It happens in `FluentProvider`
  https://github.com/microsoft/fluentui/blob/f0b05741d812ad90c52754b0b4035d5554cf2aa0/packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.ts#L96
- `FluentProvider` does not accept `ShadowRoot` in `targetDocument`
- Even if it will accept it, we need to do proper handling as changing types is not enough 🙃
